index.html

```diff
@@ -118,19 +204,55 @@
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+    if (options.party !== 'all') {
+      var party = options.party;
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+        return d.party === party;
       });
     }
 
     // SCATTERPLOT
+    var points = svg.selectAll('.point')
+      .data(txData);
```

Comment:
This is great work. One bug: I note that when I select Democrat or Republican, the circles change color, and now there are a bunch of blue circles well into Republican territory, and vice versa. What's happening here? Well we only assign the color (or technically, the class), once, right here as the circles are first entered. On subsequent updates, any existing circles will only have their position changed. But D3 doesn't know how to recognize our circles between `update()` calls, so all it can do is say, "Looks like I need 200 circles," and grab the first 200 circles, irrespective of which districts they used to represent.

One possible fix would be to assign the class in the all selection that starts on line 228. This would set the color every time `update()` is called. But we can do one better: we can teach D3 to recognize that two data points are the same by giving it a unique *key* with which to identify them. We do this here, in the `selection.data()` call. Check out the documentation for [selection.data()](https://github.com/d3/d3-3.x-api-reference/blob/master/Selections.md).